Reoranize the admin section to a more Divio-like system

nathandyer opened the pull request:
## Status

Ready for review

## Description of Changes

This PR is a first attempt at reorganizing the Admin portion of the SecureDrop documetation into a more Divio-like organizational structure. It also includes some other ancillary changes necessary for achieve this goal.

This is meant to serve as a starting point for discussion, which is greatly welcomed. I fully expect there to be a fair bit of back-and-forth and additional suggestions before it's ready to merge in (though it is in a mergeable state).

Before I describe what this PR achieves, I want to list some considerations that it intentionally does _not_ address:

### What This PR Does Not Do
* Reorganize any of the Source section of the documentation (further UX work needed)
* Reorganize any of the Journalist section of the documentation (further UX work needed)
* Create broken links to existing documentation or invalidate any prior support cases or information provided to external parties
* Add a specific section devoted to Maintenance for Admins (this is still wanted, but as that's largely new content, it's outside the scope of the PR)

### What This PR Does Do
* Makes the SecureDrop installation section more intentional about its goal of helping an end-user install and configure SecureDrop (Divio: "How-To Guide" / "Problem Oriented")
* Groups the Admin-related docs together and clarifies the intended audience, with the intention of having tutorial-related content in this section (Divio: "Tutorial"/"Learning-Oriented")
* Adds a brief section with broadly helpful information, which I have temporarily(?) named "Overview" (Divio: "Explanation"/"Understanding-Oriented")
* Eliminates the much-too-broad "Topic Guides" section
* Places all the 'best practices' documentation in the same, larger section (Divio:  "Reference"/"Information Oriented")

This partially addresses issue #417. I would vote to not close it or mark as complete until the documentation overhaul project is complete and the admin reorganization is in its "final" form.

### New Table of Contents

To make this easier for folks to contribute comments and suggestions for improvements, I'll provide a text-only version of the proposed Table of Contents:

```
Overview

    What Makes SecureDrop Unique
    Source Guide
    Journalist Guide
    Promoting Your SecureDrop Instance
    SecureDrop On-Site Training Schedule
    Getting Support

SecureDrop Installation Guide

    Overview
    Glossary
    Passphrases
    Hardware
    Before You Begin
    Create USB Boot Drives
    Set Up the Secure Viewing Station
    Set Up the Transfer Device and the Export Device
    Generate the Submission Key
    Set up the Admin Workstation
    Set Up the Network Firewall
    Setting Up a pfSense Network Firewall
    Setting Up An OPNSense Network Firewall
    Set Up the Servers
    Install SecureDrop
    Configure the Admin Workstation Post-Install and Create Backups
    Create an Admin Account on the Journalist Interface
    Test the Installation

Admin Guide

    Admin Guide
    Onboard Journalists
    Onboard Additional Admins
    Using a YubiKey with the Journalist Interface
    Investigating Logs
    OSSEC Guide
    Backing Up and Restoring Servers
    Backing Up and Restoring Workstations
    Updating Tails USBs
    Troubleshooting Kernel Updates
    HTTPS on the Source Interface
    SSH Over Local Network
    Rebuilding an Admin Workstation USB
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
    BIOS Updates on the Servers
    Off-board Administrators and Journalists
    Decommission SecureDrop

Best Practices

    Overview
    Passphrase Best Practices
    Landing Page
    Minimum requirements for the SecureDrop environment
    Whole Site Changes
    Sample SecureDrop Privacy Policy

Upgrade SecureDrop

    Upgrade from 2.5.0 to 2.5.1
    Upgrade from 2.4.2 to 2.5.0
    Upgrade from 2.4.1 to 2.4.2
    Upgrade from 2.4.0 to 2.4.1
    Upgrade from 2.3.2 to 2.4.0
    Upgrade from 2.3.1 to 2.3.2

Threat Model

    Threat Model
    Data Flow Diagram
    Attacks and Countermeasures on the SecureDrop Environment
```


## Checklist (Optional)

- [x] Doc linting (`make docs-lint`) passed locally
- [ ] Doc link linting (`make docs-linkcheck`) passed
- [x] You have previewed (`make docs`) docs at http://localhost:8000

eloquence commented:
Thanks for working on this, @nathandyer. Some initial observations from my end for further discussion:

* I would suggest keeping the user guides section untouched for now. As we discussed before, it's structurally very different from the rest of the docs, and the scope of the "Admin Guide" relative to other admin-focused sections of the documentation is poorly defined. However, structurally, it does not make sense to me to treat the very detailed journalist/source guides as part of the system overview, or to have the single-page "Admin Guide" sit on top of other chapters in a section also called "Admin Guide". 
  * Ultimately I think we'll want to unpack the "Admin Guide" and integrate it with the rest of the documentation, but I don't think we should move it around until that's done.

* I think there's a contradiction between some of the content of the "best practices" section (both in the live version and in your structure) and the framing as "best practices". For example, we refer to "minimum _requirements_" for the SecureDrop environment, and we make it clear that organizations _need to_ create a landing page. Those don't sound like "best practices" to me but explicit instructions.

* FWIW, my personal intuition for the overall structure would be: User Guides -> Overview -> Installation -> Deployment -> Maintenance -> Upgrades

nathandyer commented:
Thanks @eloquence, I appreciate that feedback! 

After thinking it over, I largely agree with your points. One ultimate goal is to break up and expand the content from the user guides into their own sections, which is why I originally put the Admin guide in a dedicated Admin section. After thinking about it further, I agree that right now it makes sense to put it alongside the other user guides, until that additional work happens.

I'm also largely in agreement about the section breakdown (for now), especially with adding a Maintenance section. I still have concerns about how we flag that as being intended for Admins (perhaps calling it "Admin Maintenance"), but it's definitely a topic worthy of its own section, especially with the planned additional content. 

Having a standalone deployment section also makes sense, although I debated with myself if some of the articles should be part of the "Deployment" section, or the "Maintenance" section. You can see where I landed on those below and in the most recent commit, but I'm still in favor of moving them around if there's general agreement that they're not in the right spots.

I do slightly disagree about where to position the "User Guides" section in the list. Despite their importance, I still think that an Overview section describing what SecureDrop is more broadly needs higher precedence in the Table of Contents. As always, I look forward to more discussion and thoughts from the team on that. 

I pushed a new commit to reflect these changes. As before, I'll provide a text version below for ease of review by the team:

```
Overview

    What Makes SecureDrop Unique
    Promoting Your SecureDrop Instance
    Passphrase Best Practices
    Threat Model
    Data Flow Diagram
    Attacks and Countermeasures on the SecureDrop Environment
    SecureDrop On-Site Training Schedule
    Getting Support

User Guides

    Source Guide
    Journalist Guide
    Admin Guide

Installation Guide

    Overview
    Glossary
    Passphrases
    Hardware
    Before You Begin
    Create USB Boot Drives
    Set Up the Secure Viewing Station
    Set Up the Transfer Device and the Export Device
    Generate the Submission Key
    Set up the Admin Workstation
    Set Up the Network Firewall
    Setting Up a pfSense Network Firewall
    Setting Up An OPNSense Network Firewall
    Set Up the Servers
    Install SecureDrop
    Configure the Admin Workstation Post-Install and Create Backups
    Create an Admin Account on the Journalist Interface
    Test the Installation

Deployment Guide

    Overview
    Landing Page
    Minimum requirements for the SecureDrop environment
    Whole Site Changes
    Sample SecureDrop Privacy Policy
    Onboard Journalists
    Onboard Additional Admins
    Using a YubiKey with the Journalist Interface
    HTTPS on the Source Interface
    SSH Over Local Network
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
    Off-board Administrators and Journalists

Maintenance Guide

    Investigating Logs
    OSSEC Guide
    Backing Up and Restoring Servers
    Backing Up and Restoring Workstations
    Updating Tails USBs
    Troubleshooting Kernel Updates
    Rebuilding an Admin Workstation USB
    BIOS Updates on the Servers
    Decommission SecureDrop

Upgrades

    Upgrade from 2.5.0 to 2.5.1
    Upgrade from 2.4.2 to 2.5.0
    Upgrade from 2.4.1 to 2.4.2
    Upgrade from 2.4.0 to 2.4.1
    Upgrade from 2.3.2 to 2.4.0
    Upgrade from 2.3.1 to 2.3.2
```

eloquence commented:
Thanks, Nathan. A couple of immediate suggestions:

- I would place "Promoting Your SecureDrop Instance" under deployment -- it's not part of an overview of the system, but relevant as part of the complete setup of an instance.
- I would consider moving "Minimum requirements" to the installation section, maybe integrated with "Before you begin".

> Having a standalone deployment section also makes sense, although I debated with myself if some of the articles should be part of the "Deployment" section, or the "Maintenance" section. 

Yes, I think that's the biggest delineation problem in this structure. I think these are the ones that are most ambiguous in where they should be placed in this structure:

```
    Using a YubiKey with the Journalist Interface
    HTTPS on the Source Interface
    SSH Over Local Network
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
```

Arguably, these guides describe _optional features_ of a SecureDrop installation. That might merit its own section between installation and deployment.

In general, I think it would be useful if we could write down a set of questions for each section that would help us decide whether a certain chapter should be placed here, e.g.:

- Installation: Do I need this information to get the environment technically up and running?
- Deployment: Do I need this information to complete a real-world SecureDrop installation?
- Maintenance: Do I need this information to keep SecureDrop running once it is set up?

nathandyer commented:
After thinking about it some more, I completely agree with the two suggestions above about moving "Promoting Your SecureDrop Instance" and "Minimum Requirements." I also agree that the latter can be folded into the "Before you begin" section, although I am concerned about making sure that information stays at the same URL (in case we've ever linked out to that in email or support correspondence).

Is there an elegant way to redirect if someone accesses a link that no longer exists, pointing to the page where the information now lives?

As for the list of ambiguous articles, I also agree that those do feel the most ambiguous. From your suggestion, I tested creating a new category on my local branch, but something about it just didn't _feel_ right. I can't say for certain why it didn't feel right, it just felt to me somewhat like an arbitrary "admin junk drawer" section. I'd love to hear others' thoughts about this, as it may just be a personal preference based on wanting to have a clear-cut install/deploy/maintain delineation (and if so, I'd be happy to ignore that internal voice and go with the additional section).

I agree that having those questions in place as a framework would help make it certain where a chapter should be placed, and I think the ones you came up with are good ones. In my mind, there's a key question that separates deployment chapters from maintenance chapters: how often does a task gets repeated?

To me, if a task is (approximately) performed one time and then doesn't need to be repeated, it's a deployment step.

If a task needs to be done multiple times, especially on a regular basis, then that seems like a maintenance step. 

With that framework in mind, those ambiguous articles seem to make the most sense to me under the Deployment section.

As always, looking forward to receiving feedback on these ideas!

nathandyer commented:
The layout as it currently stands:

```
Overview

    What Makes SecureDrop Unique
    Passphrase Best Practices
    Threat Model
    Data Flow Diagram
    Attacks and Countermeasures on the SecureDrop Environment
    SecureDrop On-Site Training Schedule
    Getting Support

User Guides

    Source Guide
    Journalist Guide
    Admin Guide

Installation Guide

    Overview
    Glossary
    Passphrases
    Hardware
    Minimum requirements for the SecureDrop environment
    Before You Begin
    Create USB Boot Drives
    Set Up the Secure Viewing Station
    Set Up the Transfer Device and the Export Device
    Generate the Submission Key
    Set up the Admin Workstation
    Set Up the Network Firewall
    Setting Up a pfSense Network Firewall
    Setting Up An OPNSense Network Firewall
    Set Up the Servers
    Install SecureDrop
    Configure the Admin Workstation Post-Install and Create Backups
    Create an Admin Account on the Journalist Interface
    Test the Installation

Deployment Guide

    Overview
    Landing Page
    Whole Site Changes
    Sample SecureDrop Privacy Policy
    Promoting Your SecureDrop Instance
    Onboard Journalists
    Onboard Additional Admins
    Using a YubiKey with the Journalist Interface
    HTTPS on the Source Interface
    SSH Over Local Network
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
    Off-board Administrators and Journalists

Maintenance Guide

    Investigating Logs
    OSSEC Guide
    Backing Up and Restoring Servers
    Backing Up and Restoring Workstations
    Updating Tails USBs
    Troubleshooting Kernel Updates
    Rebuilding an Admin Workstation USB
    BIOS Updates on the Servers
    Decommission SecureDrop

Upgrades

    Upgrade from 2.5.0 to 2.5.1
    Upgrade from 2.4.2 to 2.5.0
    Upgrade from 2.4.1 to 2.4.2
    Upgrade from 2.4.0 to 2.4.1
    Upgrade from 2.3.2 to 2.4.0
    Upgrade from 2.3.1 to 2.3.2
```

eloquence commented:
Some observations from taking another spin through it:

- Upon consideration, I would recommend keeping "Passphrase Best Practices" with the User Guides. It's essentially a user guide that applies to all three audiences, and as an instructional resource, I don't think it belongs in the overview.
- I would put "Getting Support" before "On-Site Training Schedule".
- Otherwise, I am broadly in favor of the new structure. I would recommend we do the following before merge:
    - Flesh out the "What Makes SecureDrop Unique" section a bit. If this is the first section in the documentation, I think we'll need to add a bit more basic context (what is SecureDrop?). I think this could also be an opportunity to include some of the improved illustrations that we have in our slide deck.
    - Provide a bit more context in the "Deployment" overview. Right now, the page says: "To keep sources safe, the news organization’s website, physical space, and dedicated SecureDrop hardware must employ a set of basic security best practices or risk losing any source protection provided by SecureDrop." However, the deployment guide now also contains many _optional_ considerations for a SecureDrop deployment. For this section to still make sense, we'll need to rewrite it a bit.
  - Add a bit of introductory text to the on-site training schedule. Now that you've moved it out of the proverbial junk drawer, I think we need to give it a bit more context for it to make sense in the context of the overview.

nathandyer commented:
I've gone ahead and moved "Getting Support" before the "On-Site Training" section, as suggested. I spoke with @zenmonkeykstop this morning briefly about these changes, and he was also in favor of moving the "Passphrase Best Practices," because it really doesn't make sense in its current spot. He suggested we move it to the Deployment section, although I see the recommendation above to move it to the User Guides section. Personally, it makes more sense to me with the User Guides rather than as a Deployment consideration, so I've temporarily moved it there. I'm happy to move it to Deployment if that's where we end up landing on it, of course.

I added some introductory text to the "On-Site Training Schedule", and provided some additional context in the "Deployment" overview. I look forward to reviews of that new content.

As for expanding the "What Makes SecureDrop Unique" section, I began to write that out and expand the page, but as I was writing it occurred to me that this information already existed in a fairly solid form. Turns out, I had along the way dropped the standard "Overview" from our documentation (unintentionally). 

After some more thought and consideration, I've added that section back in, with the small change that I renamed the entry to "What is SecureDrop?" That is now the opening article in the Overview section, which feels appropriate to me. In that context, the "What Makes SecureDrop Unique?" article feels right in it's current form, since it's now nested within that section. 

Does that feel like a reasonable approach here?

As always, I'll post the new layout:

```
Overview

    What Is SecureDrop?
    What Makes SecureDrop Unique?
    Threat Model
    Data Flow Diagram
    Attacks and Countermeasures on the SecureDrop Environment
    Getting Support
    SecureDrop On-Site Training Schedule

User Guides

    Source Guide
    Journalist Guide
    Admin Guide
    Passphrase Best Practices

Installation Guide

    What Is SecureDrop?
    Glossary
    Passphrases
    Hardware
    Minimum requirements for the SecureDrop environment
    Before You Begin
    Create USB Boot Drives
    Set Up the Secure Viewing Station
    Set Up the Transfer Device and the Export Device
    Generate the Submission Key
    Set up the Admin Workstation
    Set Up the Network Firewall
    Setting Up a pfSense Network Firewall
    Setting Up An OPNSense Network Firewall
    Set Up the Servers
    Install SecureDrop
    Configure the Admin Workstation Post-Install and Create Backups
    Create an Admin Account on the Journalist Interface
    Test the Installation

Deployment Guide

    Overview
    Protecting the Security of the System
    Landing Page
    Whole Site Changes
    Sample SecureDrop Privacy Policy
    Promoting Your SecureDrop Instance
    Onboard Journalists
    Onboard Additional Admins
    Using a YubiKey with the Journalist Interface
    HTTPS on the Source Interface
    SSH Over Local Network
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
    Off-board Administrators and Journalists

Maintenance Guide

    Investigating Logs
    OSSEC Guide
    Backing Up and Restoring Servers
    Backing Up and Restoring Workstations
    Updating Tails USBs
    Troubleshooting Kernel Updates
    Rebuilding an Admin Workstation USB
    BIOS Updates on the Servers
    Decommission SecureDrop

Upgrades

    Upgrade from 2.5.0 to 2.5.1
    Upgrade from 2.4.2 to 2.5.0
    Upgrade from 2.4.1 to 2.4.2
    Upgrade from 2.4.0 to 2.4.1
    Upgrade from 2.3.2 to 2.4.0
    Upgrade from 2.3.1 to 2.3.2
```

eloquence commented:
> Turns out, I had along the way dropped the standard "Overview" from our documentation (unintentionally).

In the current version of the branch, it's included twice, under the main "Overview" section and under "Installation Guide".

I think we'll probably want to refactor this a bit - a high level introduction in the "Overview" section, and a more detailed technical/operational introduction in the "Install" section. For the high level introduction, we could use the simplified diagram from our slide deck, whereas the more detailed one seems appropriate for the installation guide.

Other than that this revision looks pretty good to me! :)

nathandyer commented:
Great catch, I didn't realize I had included it twice. I've gone ahead and refactored the two overview pages significantly. I've re-titled them to be specific to their purposes. In the main overview, I've simplified some of the technical details in a few places since this is meant to be more of a high-level overview, and rearranged it to make it flow better. I also added the simplified diagram from the slide deck. In the installation overview, I've removed the extraneous pieces, directed it specifically at Admins, and have added some context for the page.

eloquence commented on docs/what_is_securedrop.rst:
```diff
@@ -180,5 +171,5 @@ to check out the :ref:`best practices <Landing Page>` for your
 :doc:`promoting your SecureDrop instance <getting_the_most_out_of_securedrop>`.
 
-.. |SecureDrop architecture overview diagram| image:: ./diagrams/SecureDrop.png
+.. |SecureDrop architecture highlevel overview diagram| image:: ./diagrams/securedrop_overview_highlevel.png
```
This diagram does not include the admin role, but it follows a paragraph that describes admins as one of the four components. I think that's potentially confusing for a reader trying to visually map the preceding paragraph to the diagram.

I would place this a bit further down (so that it's not seen as a direct visual representation of the preceding paragraph) and annotate, e.g., as follows:

> This diagram shows the process of using SecureDrop from the perspective of sources and journalists. Sources and journalists connect to SecureDrop using the Tor network (represented by the onion symbol). The SecureDrop software is running on premises on dedicated infrastructure.
>
> A source (bottom left) uploads a submission to the news organization using Tor Browser. A journalist connects to SecureDrop using their Journalist Workstation (booted from a USB drive) and physically transfers files to the air-gapped Secure Viewing Station, a machine that is never connected to the Internet. On this computer, they can process or print documents, or export them to a dedicated device.

nathandyer replied:
I agree, I made the change as requested using the annotation you provided.

eloquence commented on docs/installation_overview.rst:
```diff
@@ -0,0 +1,76 @@
+|SecureDrop architecture overview diagram|
+
+SecureDrop is a tool for sources to communicate securely with journalists. The
+SecureDrop application environment consists of three dedicated computers:
```
This section is still duplicated between the general overview and the installation overview. I would suggest keeping it only in the installation overview.

nathandyer replied:
I went back and forth on this a bit. I tested a version with the duplicate entry removed (leaving the one in the installation overview in place), but it felt like important context was missing. I think this information is critical in the general overview, but likewise, not reminding folks who are starting to step through the installation would make the process more difficult. Ultimately, I reworked this session entirely, taking out several of the duplicated paragraphs, and then re-framing the ones I kept to make sense within the context of an installation, and specifically written for an administrator. I'm curious to know your thoughts about those changes - does this shorter, more targeted version feel better to you? Or do you still think it's worth scraping one of them entirely?

eloquence commented:
Thanks @nathandyer! I've added a couple of comments re: the overview sections. 

In addition to those comments, I would suggest looking at the general overview section again from the perspective of its target audience. Does it use terms before introducing them? Would folks with different levels of technical proficiency be able to make sense of it?

I realize that you're largely using existing text here, but since we're now placing this overview upfront and are implicitly targeting multiple audiences with it, I do think we may need to give it a bit more attention with those questions in mind.

Beyond that, I think the structural changes proposed here are an improvement, and will help us approach new additions with more intentionality. I would encourage additional review from @zenmonkeykstop, and maybe @huertanix as well if he has time.

nathandyer commented:
Thank you @eloquence! I just pushed a commit with some changes to address those comments, so I'll follow up on those directly in the threads.

After looking at things again with fresher eyes, the glossary doesn't make much sense as part of the install section, because it applies to all parts of the documentation. With that in mind, I went ahead and moved it to the general overview section.

That said, having the glossary be the first part of that section is fairly clunky, so I placed it down beneath 'What Makes SecureDrop Unique?'. After reading through those first two sections again, I think they do a good job of setting everything up and providing good definitions within the context. With that in place, I feel good about having the glossary after those two introductory pages (but of course am open to alternate ways of thinking about that).

I do believe that the first half of the general overview section is fairly accessible. While there is a small assumption about what the reader understands from a technical perspective, I think any potential admin, journalist, source, or person in charge of making decisions within a newsroom would be able to grasp the concepts well. The latter parts (with the data flow diagram and the threat models) feel less accessible to me, but they make sense within the scope of the general overview since that's providing reference information. Down the line, once we have a chance to work more with UX and address expansion of source and journalist guides as well, we may find this content has another place that works even better. For now though, it feels like a good intermediate step.

I'll continue pondering on this for a bit. I look forward to seeing @zenmonkeykstop's review, as well as any others from folks who are interested. I appreciate all the great feedback so far!

--

Current layout:

```
Overview

    What Is SecureDrop?
    What Makes SecureDrop Unique?
    Glossary
    Threat Model
    Data Flow Diagram
    Attacks and Countermeasures on the SecureDrop Environment
    Getting Support
    SecureDrop On-Site Training Schedule

User Guides

    Source Guide
    Journalist Guide
    Admin Guide
    Passphrase Best Practices

Installation Guide

    Installation Overview
    Passphrases
    Hardware
    Minimum requirements for the SecureDrop environment
    Before You Begin
    Create USB Boot Drives
    Set Up the Secure Viewing Station
    Set Up the Transfer Device and the Export Device
    Generate the Submission Key
    Set up the Admin Workstation
    Set Up the Network Firewall
    Setting Up a pfSense Network Firewall
    Setting Up An OPNSense Network Firewall
    Set Up the Servers
    Install SecureDrop
    Configure the Admin Workstation Post-Install and Create Backups
    Create an Admin Account on the Journalist Interface
    Test the Installation

Deployment Guide

    Overview
    Protecting the Security of the System
    Landing Page
    Whole Site Changes
    Sample SecureDrop Privacy Policy
    Promoting Your SecureDrop Instance
    Onboard Journalists
    Onboard Additional Admins
    Using a YubiKey with the Journalist Interface
    HTTPS on the Source Interface
    SSH Over Local Network
    Accessing SecureDrop Remotely
    Setting Up a Printer in Tails
    Off-board Administrators and Journalists

Maintenance Guide

    Investigating Logs
    OSSEC Guide
    Backing Up and Restoring Servers
    Backing Up and Restoring Workstations
    Updating Tails USBs
    Troubleshooting Kernel Updates
    Rebuilding an Admin Workstation USB
    BIOS Updates on the Servers
    Decommission SecureDrop

Upgrades

    Upgrade from 2.5.0 to 2.5.1
    Upgrade from 2.4.2 to 2.5.0
    Upgrade from 2.4.1 to 2.4.2
    Upgrade from 2.4.0 to 2.4.1
    Upgrade from 2.3.2 to 2.4.0
    Upgrade from 2.3.1 to 2.3.2
```

eloquence commented:
Thanks @nathandyer. As discussed I've pushed a small commit. My goal was to make the description of the diagram a bit clearer, and to also make it work for folks with disabilities (by describing the steps in a manner that is useful even if you cannot see the diagram). Take a look and let me know what you think.

I think the overview still needs a bit of work. My main concern is that the distinction between "infrastructure" and "environment" doesn't really make sense. This predates you, but we're using it now in an introduction for all users, so I think we need to rethink it a bit.

Specifically:
- Sources, admins and journalists are not infrastructure or components. They are user roles.
- We're naming the servers (which _are_ infrastructure) both in the "infrastructure" and the "environment overview" sections.

I would suggest the following changes:
- Change the "Infrastructure" section to "user roles" and omit the servers from that section. Cross-link the relevant guides to each user role.
- Have separate "Server Infrastructure" (app server/mon server/firewall) and "Application Environment" sections to introduce the remainder

nathandyer commented:
Thank you @eloquence, those changes are a definite improvement! As for the suggestions related to "infrastructure," "environment", and "user roles," that makes total sense to me. I'll work on making those adjustments and then we can see how they feel in practice.

huertanix commented:
I don't think I'll have time to think through the re-organization of the admin docs but I can share this ticket with our edu consultant to see if she has thoughts.

eloquence reviewed (approved):
Looks good to me, thanks for all your hard work on this @nathandyer! I pushed one tiny commit, please take a quick look (7453b0b519de9388d3ce2955cd3b21ea946276a7). 

As per previous discussion, I would request that we get at least one additional maintainer's review prior to merge.

nathandyer commented:
Thank you @eloquence, that commit looks good to me! Will wait for another maintainer's review/approval before merging.

zenmonkeykstop reviewed (approved):
Re-org LGTM.
